### ISSUE TITLE
Search results not returning as expected [report by user aVNWKbpO0wc]

### ISSUE THREAD

**Issue by TBRC-pmench** [buda-base/public-digital-library#131]
This is more of a user experience issue than a bug, I think (unless there is an underlying bug in the search causing the behavior). For what it's worth, the requirement to exactly search "tsong kha pa" confused me at first as well. Other constructions do not return any results.

> The new search seems to find a bit less than the old one. For example, searching for "tsongkhapa" yielded relatively results on the old site. But if I search for "tsongkhapa" without specifying a language I get nothing, if I search for it in English I only get two results. I understand that you have now explicit search in multiple languages so maybe the expectation is to search for "tsong kha pa" in Wylie but it is something I noticed. I also noticed that I have no idea how I would search for a Sanskrit name, such as that of Chandrakirti / Candrakīrti with the new website.
I understand that the search may still be work in progress or that the behavior of the search may have changed but I wanted to mention it.

**Comment by eroux** [comment]
interesting feedback, I'm sure we'll have a good FAQ about that... @JannTibetan @TBRC-TimB  any update?

**Comment by eroux** [comment]
in any case, searching tsongkhapa on the old website indeed returns something in a note in English but it's very much random and it's not a supported way to search. We may have phonetics for the person names from ToL at some point but we don't have them now. One difference also is that the search doesn't search on the notes, it's something that we want to add, I think it shouldn't be too complex, let's see...

It would be interesting to ask the user to search Candrakīrti and Chandrakirti on the new website to see if they find the way to search in Sanskrit... I believe they'll find it but maybe I'm optimistic...

![Capture d’écran de 2020-06-29 18-18-20](https://user-images.githubusercontent.com/60868/86030791-3d23a280-ba35-11ea-82e2-7d8f021116bf.png)

**Comment by JannTibetan** [comment]
Thanks for your feedback. I’ve got many calls this morning. I’ll look into the issues in a couple hours. But yes I agree, the difference lies in the fields searched on. The system has pros and cons in this regard.

**Comment by TBRC-pmench** [comment]
@eroux 

> It would be interesting to ask the user to search Candrakīrti and Chandrakirti on the new website

I'll ask. They may be searching on the default "Version" for Candrakīrti or Chandrakirti which does produce a "No results" message (but looking to the left there are results in the facets).
![Screen Recording 2020-06-29 at 4 55 03 PM](https://user-images.githubusercontent.com/48484924/86055425-bc12de00-ba29-11ea-85d8-062c78d2911c.gif)

**Comment by eroux** [comment]
Right! You know, this makes me think that we should improve this UI... @berger-n what about this: when there's no result in Instance (or any type really), it could say:

No result found in versions, results found in other types:
- 1 work (with a link to the result page)
- 2 in persons (with a link to the results page)

(comments welcome)

**Comment by eroux** [comment]
BTW, something else that doesn't work here is that the selected type type (Version) doesn't appear in the left panel because there's no result... I think the selected type should always appear

**Comment by TBRC-pmench** [comment]
As I experiment further with this issue, I'm noticing some potentially confusing and odd behavior around search.

1. Searching Chandrakirti as Version by pressing Enter returns no results, not even as other types
2. Searching Chandrakirti as Version by selecting Sanskrit returns no results but has results appear as other types.
3. When I changed the type to person from the results page and pressed Enter, results appeared to briefly load, then returned no results.
4. Changing my search to Candrakīrti then loaded results.

![Screen Recording 2020-06-29 at 5 31 13 PM](https://user-images.githubusercontent.com/48484924/86059013-ea93b780-ba2f-11ea-8fdf-c89137ac6991.gif)

**Comment by eroux** [comment]
hmmm I think it's because if you don't select Sanskrit it will think it's Tibetan... we could change the heuristics of language detection a bit... but it will never be perfect... we may want to also indicate a bit more information to guide the user, like

`no result found in Versions for Candrakirti (searching in Tibetan Wylie)`

with perhaps an easy way to change the language... Also Nicolas, I think when the user just clicks in the search bar without typing anything, the language selection should appear. perhaps we want the language to always appear in the bar also... I don't know, there's room for improvement here, for sure. @ngawangtrinley if you want to jump in with some ideas, don't hesitate!
